website/pages/blog/scalable-apis-with-graphql-server-codegen-preset.mdx

```diff
@@ -0,0 +1,464 @@
+*   [GraphQL ESLint](https://github.com/B2o5T/graphql-eslint) for validation, linting, and checking for best practices and conventions (like all resolvers must be camel case!!!).
+
+However, there are areas to improve.
+
+For example, the guideline for one of the GraphQL servers I am working on is to use generated types from GraphQL Codegen. Some team members, particularly those new to the codebase, may have yet to be aware of this. Luckily, other team members or the Maintainers catch these issues at PR review time. But it would save everyone time and effort if we had tools to enforce this guideline automatically.
+
+### 2. How to Minimise Noise to the Maintainers
+
+Maintainers usually need to manage these aspects of a GraphQL server:
+
+*   Core server logic: resolver map, schemas, CI/CD, etc.
+*   Config files: `.graphqlrc`, `codegen.yml`, etc.
+
+Changing anything in the Maintainers' domain should notify them. Unfortunately, this happens regularly in routine workflows. Maintainers also have their team's work and general package and security updates to worry about. So, being notified of unrelated PRs quickly leads to burnout.
+
+For example, if a new resolver is added, it must be manually added to the resolver map. So, Maintainers are notified of every new resolver. It may look like this if you are using [GraphQL Yoga](https://the-guild.dev/graphql/yoga-server):
+
+```ts
+// server.ts (managed by the Maintainers)
+import { createServer } from 'node:http'
+import { createSchema, createYoga } from 'graphql-yoga'
+import { Auth } from "./schema/auth/resolvers";
+import { book, Book } from "./schema/auth/resolvers";
+import { user, User } from "./schema/user/resolvers";
+
+const schema = createSchema({
+  typeDefs: `...`,
+  resolvers: { // This is the resolver map
+    Query: {
+      user,
+      book,
+    },
+    Auth,
+    Book
+    User,
+  }
+});
+
+const yoga = createYoga({ schema });
+const server = createServer(yoga)
+server.listen(4000, () => {
+  console.info('Server is running on http://localhost:4000/graphql')
+})
+```
+
+There is a way to mitigate this issue: each schema module exports an object of resolvers, and they are merged using [mergeResolvers from @graphql-tools/merge](https://the-guild.dev/graphql/tools/docs/schema-merging#merging-resolvers). This means the Maintainers are notified on every new module instead of every new resolver:
+
+```ts
+// server.ts
+import { createServer } from 'node:http'
+import { createSchema, createYoga } from 'graphql-yoga'
+import { mergeResolvers } from '@graphql-tools/merge';
+import * as authResolvers from "./schema/auth/resolvers";
+import * as bookResolvers from "./schema/book/resolvers";
+import * as userResolvers from "./schema/user/resolvers";
+
+const schema = createSchema({
+  typeDefs: `...`,
+  resolvers: mergeResolvers([ // Resolver map is the result of merging module resolvers
```

Comment:
`createSchema` should already do the merging. You might not need `mergeResolvers` here

Reply:
TIL! Thanks for that. I have made the change [here](https://github.com/the-guild-org/website/pull/1015/commits/bb9bccfe216ae48e412b0042394303eb5a855bd8) with Laurin's suggestion